packages/web/components/table/asset-info.tsx

```diff
@@ -0,0 +1,669 @@
+    </span>
+    {usdValue && (
+      <span className="caption text-osmoverse-300">{usdValue.toString()}</span>
+    )}
+  </div>
+);
+
+const SortHeader: FunctionComponent<
+  {
+    label: string;
+    sortKey: NonNullable<SortKey>;
+    currentSortKey: SortKey;
+    currentDirection: SortDirection;
+    setSortKey: (key: SortKey) => void;
+    setSortDirection: (direction: SortDirection) => void;
+  } & CustomClasses
+> = ({
+  label,
+  sortKey,
+  currentSortKey,
+  currentDirection,
+  setSortDirection,
+  setSortKey,
+  className,
+}) => (
+  <button
+    className={classNames(
+      "ml-auto flex h-6 items-center justify-center gap-1",
+      className
+    )}
+    onClick={() => {
+      if (currentSortKey !== sortKey) {
+        // select to sort and start descending
+        setSortKey(sortKey as SortKey);
+        setSortDirection("desc");
+        return;
+      } else if (currentSortKey === sortKey && currentDirection === "desc") {
+        // toggle sort direction
+        setSortDirection("asc");
+        return;
+      } else if (currentSortKey === sortKey && currentDirection === "asc") {
+        // deselect
+        setSortKey(undefined);
+        setSortDirection("desc");
+      }
+    }}
+  >
+    <span>{label}</span>
+    {currentSortKey === sortKey && (
+      <Icon
+        width={10}
+        height={6}
+        className={classNames(
+          "ml-1 transform text-osmoverse-400 transition-transform",
+          {
+            "rotate-180": currentDirection === "asc",
+          }
+        )}
+        id="triangle-down"
+      />
+    )}
+  </button>
+);
+
+export const AssetActionsCell: AssetInfoCellComponent<{
+  onDeposit: (coinMinimalDenom: string) => void;
+  onWithdraw: (coinMinimalDenom: string) => void;
+}> = ({
+  row: {
+    original: { coinMinimalDenom },
+  },
+  onDeposit,
+  onWithdraw,
+}) => (
+  <div className="flex items-center gap-2">
+    <button
+      className="h-8 w-8 rounded-full bg-osmoverse-860 p-1"
+      onClick={() => onDeposit(coinMinimalDenom)}
+    >
+      <Icon className="m-auto" id="deposit" width={16} height={16} />
+    </button>
+    <button
+      className="h-8 w-8 rounded-full bg-osmoverse-860 p-1"
+      onClick={() => onWithdraw(coinMinimalDenom)}
+    >
+      <Icon className="m-auto" id="withdraw" width={16} height={16} />
+    </button>
+  </div>
+);
+
+const TableControls: FunctionComponent<{
+  selectedTimeFrame: CommonHistoricalPriceTimeFrame;
+  setSelectedTimeFrame: (timeFrame: CommonHistoricalPriceTimeFrame) => void;
+  setSearchQuery: (searchQuery: Search | undefined) => void;
+  selectedView: "myTokens" | "allTokens";
+  setSelectedView: (view: "myTokens" | "allTokens") => void;
+}> = ({
+  selectedTimeFrame,
+  setSelectedTimeFrame,
+  setSearchQuery,
+  selectedView,
+  setSelectedView,
+}) => {
+  const { t } = useTranslation();
+
+  const [searchInput, _, setSearchInput, searchQueryInput] =
```

Comment:
Thoughts on making this an object instead of an array? It's faster to get the properties through autocomplete than Typescript hover.